sound/soc/sof/pm.c

```diff
@@ -359,11 +359,16 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 #endif
 	/* notify DSP of upcoming power down */
 	ret = sof_send_pm_ipc(sdev, SOF_IPC_PM_CTX_SAVE);
-	if (ret < 0) {
+	if (ret == -EBUSY || ret == -EAGAIN) {
```

Comment:
@kv2019i for my education, when can this happen? And what would the PM core do, retry at a later point or just not suspend?

Reply:
This depends on the SOF backend implementation. Our current Intel backends never return EBUSY/EAGAIN, but given that SOF interface closely follows the Linux PM API, I think we should allow this for backends.

On Linux side, with S3-suspend, this will just abort the suspend process.

With runtime-suspend, suspend will be attempted later. The behaviour is documented as follows: "" If ->runtime_suspend failed with -EAGAIN or -EBUSY, and if the RPM_AUTO flag is set and the next autosuspend-delay expiration time is in the future, schedule another autosuspend attempt.""

Reply:
ok, please add a comment on this so that there is a trace of the design in the code - this is not a random set of codes. thanks!

Reply:
@plbossart, ack, addressed now